apps/studio/components/interfaces/QueryPerformanceV2/QueryPerformance.tsx

```diff
@@ -0,0 +1,210 @@
+  Tabs_Shadcn_,
+  TooltipContent_Shadcn_,
+  TooltipTrigger_Shadcn_,
+  Tooltip_Shadcn_,
+  cn,
+} from 'ui'
+import ConfirmModal from 'ui-patterns/Dialogs/ConfirmDialog'
+import ShimmeringLoader from 'ui-patterns/ShimmeringLoader'
+import { Markdown } from '../Markdown'
+import { useQueryPerformanceQuery } from '../Reports/Reports.queries'
+import { PresetHookResult } from '../Reports/Reports.utils'
+import { QUERY_PERFORMANCE_REPORT_TYPES } from './QueryPerformance.constants'
+import { QueryPerformanceFilterBar } from './QueryPerformanceFilterBar'
+import { QueryPerformanceGrid } from './QueryPerformanceGrid'
+
+interface QueryPerformanceProps {
+  queryHitRate: PresetHookResult
+  queryPerformanceQuery: DbQueryHook<any>
+}
+
+export const QueryPerformance = ({
+  queryHitRate,
+  queryPerformanceQuery,
+}: QueryPerformanceProps) => {
+  const router = useRouter()
+  const { preset } = useParams()
+  const { project } = useProjectContext()
+  const [page, setPage] = useState<QUERY_PERFORMANCE_REPORT_TYPES>(
+    (preset as QUERY_PERFORMANCE_REPORT_TYPES) ?? QUERY_PERFORMANCE_REPORT_TYPES.MOST_TIME_CONSUMING
+  )
+  const [showResetgPgStatStatements, setShowResetgPgStatStatements] = useState(false)
+
+  const handleRefresh = () => {
+    queryPerformanceQuery.runQuery()
+    queryHitRate.runQuery()
+  }
+
+  const { data: mostTimeConsumingQueries, isLoading: isLoadingMTC } = useQueryPerformanceQuery({
+    preset: 'mostTimeConsuming',
+  })
+  const { data: mostFrequentlyInvoked, isLoading: isLoadingMFI } = useQueryPerformanceQuery({
+    preset: 'mostFrequentlyInvoked',
+  })
+  const { data: slowestExecutionTime, isLoading: isLoadingMMF } = useQueryPerformanceQuery({
+    preset: 'slowestExecutionTime',
+  })
+
+  const QUERY_PERFORMANCE_TABS = [
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.MOST_TIME_CONSUMING,
+      label: 'Most time consuming',
+      description: 'Lists queries ordered by their cumulative total execution time.',
+      isLoading: isLoadingMTC,
+      max:
+        (mostTimeConsumingQueries ?? []).length > 0
+          ? Math.max(...(mostTimeConsumingQueries ?? []).map((x: any) => x.total_time)).toFixed(2)
+          : undefined,
+    },
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.MOST_FREQUENT,
+      label: 'Most frequent',
+      description: 'Lists queries in order of their execution count',
+      isLoading: isLoadingMFI,
+      max:
+        (mostFrequentlyInvoked ?? []).length > 0
+          ? Math.max(...(mostFrequentlyInvoked ?? []).map((x: any) => x.calls)).toFixed(2)
+          : undefined,
+    },
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.SLOWEST_EXECUTION,
+      label: 'Slowest execution',
+      description: 'Lists queries ordered by their maximum execution time',
+      isLoading: isLoadingMMF,
+      max:
+        (slowestExecutionTime ?? []).length > 0
+          ? Math.max(...(slowestExecutionTime ?? []).map((x: any) => x.max_time)).toFixed(2)
+          : undefined,
+    },
+  ]
```

Comment:
Might be a pre-optimisation, but since we're using `Math.max()` in here it might be worth wrapping this in a `useMemo()`